dart/lib/flat_buffers.dart

```diff
@@ -353,6 +355,14 @@ class Builder {
             fileIdentifier.codeUnitAt(i));
       }
     }
+
+    // zero out the added padding
+    for (var i = sizeBeforePadding + 1;
+        i <= finishedSize - requiredBytes;
+        i++) {
+      _setUint8AtTail(_buf, i, 0);
+    }
```

Comment:
Dart zero initializes ByteBuffers by default. I don't think this is the right place to do this. We should just document that custom allocators need to do this, no?